tools/info/main.cpp

```diff
@@ -222,7 +224,16 @@ std::string GetJsonValue(const nlohmann::json& json_obj, const std::string& key)
 
     if (json_obj.contains(key))
     {
-        out = json_obj.at(key);
+        try
+        {
+            out = json_obj.at(key).get<std::string>();
+        }
+        catch (const nlohmann::json::type_error& te)
+        {
+            out += "\n\t" + json_obj.at(key).dump(kDefaultIndent);
+            out.pop_back();
+            out += "\t}";
+        }
```

Comment:
The use of try-catch here feels awkward, as it should be possible to know if its a string or not ahead of time.

Should be possible to call `is_string()` before printing, and using the fallback if it isn't a string.

https://json.nlohmann.me/api/basic_json/is_string/

I say this because other uses of try-catch in the codebase are in main() functions, or are handling potentially invalid input (like stof), whereas here there is a way to "know" what to do without relying on an exception to do so.

Reply:
Oh yes, somehow I missed this method while I was reading the documentation. Your suggestion makes much more sense. Furthermore, thanks to your comment I revisited this piece of code, and made further adjustments.